applications_superstaq/logistics.py

```diff
@@ -56,3 +57,65 @@ def read_json_warehouse(json_dict: dict) -> WarehouseOutput:
     return WarehouseOutput(
         warehouse_to_destination, total_distance, map_link, open_warehouses, qubo
     )
+
+
+class Logistics:
+    def __init__(self, client: superstaq_client._SuperstaQClient):
+        self._client = client
+
+    def tsp(self, locs: List[str], solver: str = "anneal") -> TSPOutput:
+        """
+        This function solves the traveling salesperson problem (TSP) and
+        takes a list of strings as input. TSP finds the shortest tour that
+        traverses all locations in a list.
+        Each string should be an addresss or name of a landmark
+        that can pinpoint a location as a Google Maps search.
+        It is assumed that the first string in the list is
+        the starting and ending point for the TSP tour.
+        The function returns a dictionary containing the route,
+        the indices of the route from the input list, and the total distance
+        of the tour in miles.
+        Args:
+            locs: List of strings where each string represents
+            a location needed to be visited on tour.
+            solver: A string indicating which solver to use ("rqaoa" or "anneal").
+        Returns:
+            A TSPOutput object with the following attributes:
+            .route: The optimal TSP tour as a list of strings in order.
+            .route_list_numbers: The indicies in locs of the optimal tour.
+            .total_distance: The tour's total distance.
+            .map_links: A link to google maps that show the tour.
+        """
+        input_dict = {"locs": locs}
+        json_dict = self._client.tsp(input_dict)
+
+        return read_json_tsp(json_dict)
+
+    def warehouse(
+        self, k: int, possible_warehouses: List[str], customers: List[str], solver: str = "anneal"
+    ) -> WarehouseOutput:
+        """
+        This function solves the warehouse location problem, which is:
+        given a list of customers to be served and  a list of possible warehouse
+        locations, find the optimal k warehouse locations such that the sum of
+        the distances to each customer from the nearest facility is minimized.
+        Args:
+            k: An integer representing the number of warehouses in the solution.
+            possible_warehouses: A list of possible warehouse locations.
+            customers: A list of customer locations.
+            solver: A string indicating which solver to use ("rqaoa" or "anneal").
+        Returns:
+            A WarehouseOutput object with the following attributes:
+            .warehouse_to_destination: The optimal warehouse-customer pairings in List(Tuple) form.
+            .total_distance: The tour's total distance among all warehouse-customer pairings.
```

Comment:
what is the "tour" in this case?

Reply:
The sum of the distances to each customer from the nearest facility.

Reply:
ok, I don't think this is super clear from preceding text. can it just be left out?
```suggestion
            .total_distance: The total distance among all warehouse-customer pairings.
```

Reply:
Done.